File: repository/src/main/java/com/bernatgomez/apps/randomuser/models/DataUser.java
package com.bernatgomez.apps.randomuser.models;

/**
 * Created by bernatgomez on 08/09/2017.
 */

public class DataUser extends BaseData {

    public DataUser(String name) {
        this.setName(name);
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    private String name = "";
}

File: data/src/main/java/com/bernatgomez/apps/randomuser/models/DataUser.kt
package com.bernatgomez.apps.randomuser.models


/**
 * API abstraction for random user elements
 *
 * Created by bernatgomez on 09/09/2017.
 */
data class DataUser constructor(
        val gender : String, val name : DataName, val location : DataLocation, val email : String, val login : DataLogin,
        val dob : String, val registered : String, val phone : String, val cell : String, val id : DataId, val picture : DataPicture,
        val nat : String
        ) {

        constructor() : this("", DataName(), DataLocation(), "", DataLogin(), "", "", "", "", DataId(), DataPicture(), "")

}